t/bc/bc.t

```diff
@@ -0,0 +1,207 @@
+        [ '7 > 5',        '1',       'greater than' ],
+        [ '7 >= 7',       '1',       'greater than or equal' ],
+    ];
+    return $table;
+}
+
+sub precedence_table {
+    my $table = [
+        'precedence',
+        [ 'v = (3 < 5)',  '1',       'just like gnu bc' ],
+
+        # according to the GNU bc documentation, their implementation
+        # will assign 3 to v and then do the relational test
+        [ 'v = 3 < 5',    '1',       ' not like gnu bc' ],
+
+        [ 'v = 4+5*2^3',  '44',      'PEDMAS' ],
+        [ 'v = (4+5)*2',  '18',      'PEDMAS' ],
+        [ 'v = -(4+5)+8', '-1',      'PEDMAS' ],
+    ];
+    return $table;
+}
+
+sub special_expr_table {
+    my $sqrt2 = '1.4142135623731';
+    my $table = [
+        'special expressions',
+        [ 'sqrt(2)',          $sqrt2,     'sqrt function' ],
+        [ 'length(sqrt(2))',  '14',       'significant digits' ],
+        [ 'scale(sqrt(2))',   '13',       'precision digits' ],
+    ];
+    return $table;
+}
+
+sub statement_table {
+    ## no critic [ValuesAndExpressions::RequireInterpolationOfMetachars]
+    my $table = [
+        'statements',
+        [    q("a string"),
+            'a string',
+            'string literal',
+        ],
+
+        [
+            # need to print the newline here so that the test harness
+            # recognizes the end of the test output
+            'print "1+2", " is ", 1+2, "\n"',
+            '1+2 is 3',
+            'print statement',
+        ],
+
+        [   '{ print "a"; print "b"; print "\n" }',
+            'ab',
+            'compound statement',
+        ],
+
+        [   'v=5; while (v--) { print v }; print "\n"',
+            "5\n43210",
+            'while statement',
+        ],
+
+        ## no critic [ValuesAndExpressions::ProhibitInterpolationOfLiterals]
+        [   'for (v=0;v<5;v++) { print v }; print "\n"',
+            "01234",
+            'for statement',
+        ],
+
+        [   'for ( v=0; v<5; v++) { print v; if (v>2) break }; print "\n"',
+            "01230",
+            'for with break statement',
+        ],
+
+        # The following aren't being tested because they are extensions
+        # that aren't support in the PerlPowerTools implementation of bc
+        # - read()
+        # - if ... else
+        # - continue
+        # - halt
+        # - warranty
+        # - define (because I can't even get it to work interactively)
+
+        # Not testing this, for obvious reasons
+        # - quit
+    ];
+    return $table;
+}
+
+sub run_table {
+    my ($table) = @_;
+
+    my $label = shift @$table;
+
+    subtest $label => sub {
+        my ( $fh, $input ) = tempfile();
+        foreach my $tuple (@$table) {
+            my ( $expr, $expected, $desc ) = @$tuple;
+            print {$fh} "$expr\n" or die;
+        }
+        close $fh or die;
+
+        ## no critic [InputOutput::ProhibitBacktickOperators]
+        my $output = `perl $Script $input`;
```

Comment:
Don't use "perl" here. Use `"$^X" ...` to use the currently running perl, with the quotes there to protect whitespace in the path.

Reply:
Yeah I noticed that but it's something I don't usually do and so I didn't think of it at the time.  I'll fix it.

BTW, is there any reason why you don't use FindBin the test scripts?  It's a bit of a nuisance being restricted to running the script from the top folder.

Reply:
Change pushed.